### PR TITLE
onRenderCoin to be called irrespective of  imageUrl being present or not

### PR DESCRIPTION
**Pull request by s-vaibhav**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

The callback onRenderCoin, if passed, will be called irrespective of imageUrl being present or not.
This will further decide to render the Image component or not based on imageUrl being present or not.

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DocumentCard/__snapshots__/DocumentCard.test.tsx.snap [hunk 1]
```diff
@@ -95,7 +95,6 @@ exports[`DocumentCard renders DocumentCard correctly 1`] = `
                 
               </i>
             </div>
-
```
Comment (mtennoe) [inline packages/office-ui-fabric-react/src/components/DocumentCard/__snapshots__/DocumentCard.test.tsx.snap]: Unrelated?

File: packages/office-ui-fabric-react/src/components/Persona/PersonaCoin/PersonaCoin.base.tsx [hunk 2]
```diff
@@ -161,7 +161,9 @@ export class PersonaCoinBase extends BaseComponent<IPersonaCoinProps, IPersonaSt
       showUnknownPersonaCoin
     });
 
-    return (
+    // return Image if imageUrl is present
+    // this check will stop from rendering an empty img tag
```
Comment (mtennoe) [inline packages/office-ui-fabric-react/src/components/Persona/PersonaCoin/PersonaCoin.base.tsx]: Some typos. What about:
"Render the Image component only if an image URL is provided"?

File: packages/office-ui-fabric-react/src/components/Persona/PersonaCoin/PersonaCoin.base.tsx [hunk 3]
```diff
@@ -173,7 +175,7 @@ export class PersonaCoinBase extends BaseComponent<IPersonaCoinProps, IPersonaSt
         shouldStartVisible={imageShouldStartVisible}
         onLoadingStateChange={this._onPhotoLoadingStateChange}
       />
-    );
+    ) : null;
```
Comment (mtennoe) [inline packages/office-ui-fabric-react/src/components/Persona/PersonaCoin/PersonaCoin.base.tsx]: >null [](start = 8, length = 4)

We can move the null imageUrl check to the top of the function and return null immediately in that case

File: packages/office-ui-fabric-react/src/components/DocumentCard/__snapshots__/DocumentCard.test.tsx.snap (hunk 1 above)
Comment (s-vaibhav) [reply packages/office-ui-fabric-react/src/components/DocumentCard/__snapshots__/DocumentCard.test.tsx.snap]: Yes

**Comment by mtennoe** [comment]
Can we add some tests for ensuring that this doesnt regress again? I.e. a test that expects onRenderCoin to be called without an imageUrl provided?

File: packages/office-ui-fabric-react/src/components/Persona/Persona.test.tsx [hunk 4]
```diff
@@ -22,10 +22,15 @@ const STYLES = {
  * function to override the default onRender callbacks
  */
 export const wrapPersona = (
-  example: IPersonaSharedProps
+  example: IPersonaSharedProps,
+  isCoin: boolean = false
 ): ((coinProps: IPersonaProps, defaultRenderer: IRenderFunction<IPersonaProps>) => JSX.Element | null) => {
   return (coinProps, defaultCoinRenderer): JSX.Element | null => {
-    return defaultCoinRenderer(coinProps);
+    return isCoin ? (
```
Comment (mtennoe) [inline packages/office-ui-fabric-react/src/components/Persona/Persona.test.tsx]: From the name, it is unclear to me what "isCoin" means in this case. Would prefer a name that explains what the result of setting the parameter would be

File: packages/office-ui-fabric-react/src/components/Persona/Persona.test.tsx [hunk 5]
```diff
@@ -23,10 +23,10 @@ const STYLES = {
  */
 export const wrapPersona = (
   example: IPersonaSharedProps,
-  isCoin: boolean = false
+  isOnRenderCoinCB: boolean = false
```
Comment (mtennoe) [inline packages/office-ui-fabric-react/src/components/Persona/Persona.test.tsx]: Would prefer typing out CB => callback.
Even better, what about shouldWrapPersonaCoin? That explains exactly what is happening

**Comment by mtennoe** [comment]
@s-vaibhav - Can you try reverting the unrelated file changes (the whitespace ones) to reduce the list of codeowners needed to approve this?
